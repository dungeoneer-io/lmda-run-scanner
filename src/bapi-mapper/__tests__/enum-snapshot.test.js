const { describe, expect, test } = require('@jest/globals');

const {
    mapApiToRaces,
    mapApiToSpecs
} = require('../enum-snapshot');

const ArrayOfRaces = require('./__fixtures__/array-of-races.json');
const ArrayOfSpecs = require('./__fixtures__/array-of-specs.json');

describe('transformers', () => {
    test('mapApiToRaces', () => {
        const actual = ArrayOfRaces.map(mapApiToRaces);
        expect(actual).toMatchSnapshot();
    });

    test('mapApiToSpecs', () => {
        const result = ArrayOfSpecs.map(mapApiToSpecs);
        expect(result).toMatchSnapshot();
    });
});