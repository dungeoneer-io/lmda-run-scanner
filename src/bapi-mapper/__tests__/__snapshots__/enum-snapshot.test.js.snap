// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformers mapApiToRaces 1`] = `
[
  {
    "faction": 0,
    "id": 26,
    "isAllied": false,
    "name": "Pandaren",
  },
  {
    "faction": 1,
    "id": 25,
    "isAllied": false,
    "name": "Pandaren",
  },
  {
    "faction": 2,
    "id": 24,
    "isAllied": false,
    "name": "Pandaren",
  },
]
`;

exports[`transformers mapApiToSpecs 1`] = `
[
  {
    "class": 1,
    "id": 73,
    "name": "Protection",
    "role": 1,
  },
  {
    "class": 1,
    "id": 72,
    "name": "Fury",
    "role": 3,
  },
]
`;
